app/src/full/kotlin/io/homeassistant/companion/android/settings/wear/SettingsWearActivity.kt

```diff
@@ -108,97 +102,32 @@ class SettingsWearActivity :
-        try {
-            val capabilityInfo = capabilityClient
-                ?.getCapability(CAPABILITY_WEAR_APP, CapabilityClient.FILTER_ALL)
-                ?.await()
-
-            withContext(Dispatchers.Main) {
-                wearNodesWithApp = capabilityInfo?.nodes
-                Timber.d("Capable Nodes: $wearNodesWithApp")
-                updateUI()
-            }
-        } catch (cancellationException: CancellationException) {
-            // Request was cancelled normally
-            throw cancellationException
-        } catch (throwable: Throwable) {
-            Timber.d("Capability request failed to return any results.")
-        }
-    }
-
-    private suspend fun findAllWearDevices() {
-        try {
-            val connectedNodes = nodeClient?.connectedNodes?.await()
-
-            withContext(Dispatchers.Main) {
-                allConnectedNodes = connectedNodes
-                updateUI()
-            }
-        } catch (cancellationException: CancellationException) {
-            // Request was cancelled normally
-        } catch (throwable: Throwable) {
-            Timber.d("Node request failed to return any results.")
-        }
-    }
-
-    private fun updateUI() {
-        val wearNodesWithApp = wearNodesWithApp
-        val allConnectedNodes = allConnectedNodes
-
-        when {
-            wearNodesWithApp == null || allConnectedNodes == null -> {
-                Timber.d("Waiting on Results for both connected nodes and nodes with app")
-                binding.informationTextView.text = getString(commonR.string.message_checking)
-                binding.remoteOpenButton.isInvisible = true
-            }
-            allConnectedNodes.isEmpty() -> {
-                Timber.d("No devices")
-                binding.informationTextView.text = getString(commonR.string.message_no_connected_nodes)
-                binding.remoteOpenButton.isInvisible = true
-            }
-            wearNodesWithApp.isEmpty() -> {
-                Timber.d("Missing on all devices")
-                binding.informationTextView.text = getString(commonR.string.message_missing_all)
-                binding.remoteOpenButton.isVisible = true
-            }
-            wearNodesWithApp.size < allConnectedNodes.size -> {
-                Timber.d("Installed on some devices")
-                startActivity(
-                    SettingsWearMainView.newInstance(applicationContext, wearNodesWithApp, getAuthIntentUrl()),
-                )
-                finish()
-            }
-            else -> {
-                Timber.d("Installed on all devices")
-                startActivity(
-                    SettingsWearMainView.newInstance(applicationContext, wearNodesWithApp, getAuthIntentUrl()),
-                )
-                finish()
-            }
-        }
+    private fun openWearDeviceSettings() {
+        val wearNodesWithApp = settingsWearViewModel.wearNodesWithApp.value ?: return
```

Comment:
Careful here you need to check if it's not empty since it cannot be null anymore.